### PR TITLE
Convert testarch to graph2 library.

### LINKED ISSUE #486: xc7 xml fails to lint
`make artix7_xc7a50t_test_rrxml_real_lint` fails

<details>
<summary>Traceback</summary>

```
Generating rr_graph_xc7a50t_test.rr_graph.real.lint.html
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': [facet 'enumeration'] The value 'NO_SIDE' is not an element of the set {'LEFT', 'RIGHT', 'TOP', 'BOTTOM'}.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': 'NO_SIDE' is not a valid value of the atomic type 'loc_side'.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': [facet 'enumeration'] The value 'NO_SIDE' is not an element of the set {'LEFT', 'RIGHT', 'TOP', 'BOTTOM'}.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': 'NO_SIDE' is not a valid value of the atomic type 'loc_side'.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': [facet 'enumeration'] The value 'NO_SIDE' is not an element of the set {'LEFT', 'RIGHT', 'TOP', 'BOTTOM'}.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': 'NO_SIDE' is not a valid value of the atomic type 'loc_side'.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': [facet 'enumeration'] The value 'NO_SIDE' is not an element of the set {'LEFT', 'RIGHT', 'TOP', 'BOTTOM'}.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': 'NO_SIDE' is not a valid value of the atomic type 'loc_side'.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': [facet 'enumeration'] The value 'NO_SIDE' is not an element of the set {'LEFT', 'RIGHT', 'TOP', 'BOTTOM'}.
/home/elms/work/symbiflow/symbiflow-arch-defs/build/xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.xml:2: element loc: Schemas validity error : Element 'loc', attribute 'side': 'NO_SIDE' is not a valid value of the atomic type 'loc_side'.
xc7/archs/artix7/devices/CMakeFiles/artix7_xc7a50t_test_rrxml_real_lint.dir/build.make:62: recipe for target 'xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.lint.html' failed
make[3]: *** [xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.lint.html] Error 141
make[3]: *** Deleting file 'xc7/archs/artix7/devices/rr_graph_xc7a50t_test.rr_graph.real.lint.html'
CMakeFiles/Makefile2:32260: recipe for target 'xc7/archs/artix7/devices/CMakeFiles/artix7_xc7a50t_test_rrxml_real_lint.dir/all' failed
make[2]: *** [xc7/archs/artix7/devices/CMakeFiles/artix7_xc7a50t_test_rrxml_real_lint.dir/all] Error 2
CMakeFiles/Makefile2:32267: recipe for target 'xc7/archs/artix7/devices/CMakeFiles/artix7_xc7a50t_test_rrxml_real_lint.dir/rule' failed
make[1]: *** [xc7/archs/artix7/devices/CMakeFiles/artix7_xc7a50t_test_rrxml_real_lint.dir/rule] Error 2
Makefile:10354: recipe for target 'artix7_xc7a50t_test_rrxml_real_lint' failed
make: *** [artix7_xc7a50t_test_rrxml_real_lint] Error 2
```

</details>

### PR DESCRIPTION
**Pull request by litghost**
Removes use of graph (per #339) and lays groundwork for constant routing experiment.

This also exposed some bugs in some paths of the graph2 library, which are fixed.

Also fixes #486

### CODE REVIEW

File: testarch/utils/testarch_graph.py [hunk 1]
```diff
@@ -1,140 +1,174 @@
...
-            try_connect(ytrack, xtrack)
-
+    # Iterate over all valid x channels and connect to all valid y channels
+    # and vice versa as direction implies
+
+    def try_connect(ainode, binode):
+        atrack = graph.nodes[ainode]
+        btrack = graph.nodes[binode]
+        # One of the nodes should be going in and the other should be going out
+        # Filter out grossly non-sensical connections that confuse the VPR GUI
```
[the remark this answers is not in the document]
Comment (litghost) [reply testarch/utils/testarch_graph.py]: That comment and the algorithm was a direct copy from the previous implementation.

File: testarch/utils/testarch_graph.py [hunk 2]
```diff
@@ -1,140 +1,174 @@
...
-            if x != grid_sz.width - 1:
-                tracks[graph.RoutingNodeSide.RIGHT] = g.channels.y.column(x)[
-                    1][tracki]
-            # Now bind to all adjacent pins
-            for block in g.block_grid.blocks_for(col=x):
-                connect_block_to_track(block, tracks)
-
-
-def connect_tracks_to_tracks(g, switch, verbose=False):
+    for loc in graph.grid:
```
[the remark this answers is not in the document]
Comment (litghost) [reply testarch/utils/testarch_graph.py]: This operation is something that no real arch actually needs to do, it is something that only the test arch needs.  Real architecture do not do this all-to-all style connection, instead they just make connections based on existing pips.

File: testarch/utils/testarch_graph.py [hunk 3]
```diff
@@ -1,140 +1,174 @@
...
+
+    def try_connect(ainode, binode):
+        atrack = graph.nodes[ainode]
+        btrack = graph.nodes[binode]
+        # One of the nodes should be going in and the other should be going out
+        # Filter out grossly non-sensical connections that confuse the VPR GUI
+        if (btrack.direction == graph2.NodeDirection.INC_DIR
+                and channel_start(btrack) <= channel_common(atrack)) \
+            or (btrack.direction == graph2.NodeDirection.DEC_DIR
+                and channel_start(btrack) > channel_common(atrack)):
```
Comment (elmsfu) [inline testarch/utils/testarch_graph.py]: I know it's just a translation of the previous implementation
```suggestion
                and channel_start(btrack) >= channel_common(atrack)):
```
Not sure why you connect `DEC_DIR` only from either x->y or y->x exclusively for non-square grids.

File: testarch/utils/testarch_graph.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (litghost) [reply testarch/utils/testarch_graph.py]: I've mode the pin walk to a function.

File: testarch/utils/testarch_graph.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (litghost) [reply testarch/utils/testarch_graph.py]: I tried to re-write it.

File: testarch/utils/testarch_graph.py [hunk 4]
```diff
@@ -1,140 +1,174 @@
...
+                        direction='Y',
+                        x_low=begin[0],
+                        x_high=end[0],
+                        y_low=begin[1],
+                        y_high=end[1],
+                    ),
+                segment_id=graph.segments[0].id,
+                capacity=1,
+                direction=direction,
+                name="CHANY{:04d}@{:04d}".format(y, tracki),
```
[the remark this answers is not in the document]
Comment (litghost) [reply testarch/utils/testarch_graph.py]: Fixed

File: testarch/utils/testarch_graph.py (hunk 3 above)
Comment (litghost) [reply testarch/utils/testarch_graph.py]: Totally valid, fixed.

**Pull request merged**